pkg/asset/installconfig/gcp/client.go

```diff
@@ -48,6 +49,36 @@ func NewClient(ctx context.Context) (*Client, error) {
 	return client, nil
 }
 
+// GetMachineTypes uses the GCP Compute Service API to get a list of machine types from a project
+func (c *Client) GetMachineTypes(ctx context.Context, project, filter string) (map[string]*compute.MachineType, error) {
+	types := map[string]*compute.MachineType{}
+
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	svc, err := c.getComputeService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := svc.MachineTypes.AggregatedList(project)
+	if filter != "" {
+		req = req.Filter(filter)
+	}
+
+	if err := req.Pages(ctx, func(page *compute.MachineTypeAggregatedList) error {
+		for _, scopedList := range page.Items {
+			for _, item := range scopedList.MachineTypes {
+				types[item.Name] = item
```

Comment:
shouldn't this nested lookup exit earlier whenever possible?

Reply:
@bogdando What are you suggesting as criteria for existing early? The expectation is that the code will add **all** of the machine types to the `types` map.

Reply:
No, this is designed to allow multiple machine types to be retrieved. It is true that it will experience several of the same type (one for each zone, in each region). However, if we exited here then we would miss all of the other types.